File: cpp/35_搜索插入位置_easy.cpp
// 按部就班 
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        if(nums.size()==0)
            return 0;
        for(int i=0; i<nums.size(); i++)
            if(nums[i]==target)
                return i;
        int index=0;
        while(index<nums.size() && nums[index]<target)
            index++;
        return index;
    }
};

// 二分查找
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int N = nums.size();
        if(N==0) return 0;
        int low=0, high=N-1;
        while(low<=high){
            int mid = (low+high)/2;
            if(target==nums[mid])
                return mid;
            else if(target<nums[mid])
                high = mid-1;
            else low = mid+1;
        }
        return low;
    }
};

File: python/35_搜索插入位置_easy.py
# 按部就班
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if not nums:
            return 0;
        for i in range(len(nums)):
            if nums[i]==target:
                return i
        index = 0
        while index<len(nums) and nums[index]<target:
            index += 1
        return index

# 二分法:
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        N = len(nums)
        low, high = 0, N-1
        while low<=high:
            mid = (low+high)>>1
            if target==nums[mid]:
                return mid
            elif target<nums[mid]:
                high = mid - 1
            else:
                low = mid + 1
        return low